src/Umbraco.Cms.Api.Management/Controllers/Document/AllowedChildrenOfRootDocumentController.cs

```diff
@@ -0,0 +1,41 @@
+﻿using Asp.Versioning;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Umbraco.Cms.Api.Common.ViewModels.Pagination;
+using Umbraco.Cms.Api.Management.ViewModels.DocumentType;
+using Umbraco.Cms.Core.Mapping;
+using Umbraco.Cms.Core.Models;
+using Umbraco.Cms.Core.Services;
+
+namespace Umbraco.Cms.Api.Management.Controllers.Document;
+
+public class AllowedChildrenOfRootDocumentController : DocumentControllerBase
+{
+    private readonly IContentTypeService _contentTypeService;
+    private readonly IUmbracoMapper _umbracoMapper;
+
+    public AllowedChildrenOfRootDocumentController(IContentTypeService contentTypeService, IUmbracoMapper umbracoMapper)
+    {
+        _contentTypeService = contentTypeService;
+        _umbracoMapper = umbracoMapper;
+    }
+
+    [HttpGet("root/allowed-document-types")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(PagedViewModel<DocumentTypeResponseModel>), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> AllowedChildrenOfRoot(int skip = 0, int take = 100)
+    {
+        IEnumerable<IContentType> allowedChildrenOfRoot = _contentTypeService.GetAll().Where(x => x.AllowedAsRoot).ToArray();
+
+        List<DocumentTypeResponseModel> viewModels = _umbracoMapper.MapEnumerable<IContentType, DocumentTypeResponseModel>(allowedChildrenOfRoot.Skip(skip).Take(take));
```

Comment:
We should move the functionality about skip take into a service

Reply:
Should be fixed 👍